## Part 1: Explore the Data

Import the data and use Pandas to learn more about the dataset.

In [1]:
import pandas as pd

df = pd.read_csv('Resources/client_dataset.csv')
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year         item_id     category         subcategory  \
0          17        2023  EUD29711-63-6U        decor            wall art   
1          20        2023  XDA18116-89-4A  consumables                pens   
2           4        2023  ABE59463-05-7E     software  project management   
3          17        2023  ZMM00836-65-0C  consumables                pens   
4          21        2023  BZX55559-12-3X  consumables                misc   

   unit_price  unit_cost  unit_weight  qty  line_number  
0     1096.80     762.71         7.50  105            1  
1       24.95      15.09         1.49   21            0  
2       13.52       7.86         1.68   39            6  
3       36.42      24.85         1.23   29            3  
4      195.10     108.17        46.43   20            1

In [2]:
# View the column names in the data
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number'],
      dtype='object')

In [3]:
# Use the describe function to gather some basic statistics
df.describe()

client_id      order_id    order_week    order_year    unit_price  \
count  54639.000000  5.463900e+04  54639.000000  54639.000000  54639.000000   
mean   54837.869416  5.470190e+06     11.359139   2022.993064    136.267207   
std    25487.438231  2.599807e+06      7.023499      0.082997    183.873135   
min    10033.000000  1.000886e+06      1.000000   2022.000000      0.010000   
25%    33593.000000  3.196372e+06      6.000000   2023.000000     20.800000   
50%    53305.000000  5.496966e+06     11.000000   2023.000000     68.310000   
75%    78498.000000  7.733869e+06     17.000000   2023.000000    173.160000   
max    99984.000000  9.998480e+06     52.000000   2023.000000   1396.230000   

          unit_cost   unit_weight           qty   line_number  
count  54639.000000  54639.000000  5.463900e+04  54639.000000  
mean      99.446073      5.004116  5.702646e+02      2.979667  
std      133.164267      5.326599  1.879552e+04      2.436320  
min        0.010000      0.000000  0.000000e+00      0.000000  
25%       14.840000      1.450000  3.200000e+01      1.000000  
50%       49.890000      3.240000  6.800000e+01      3.000000  
75%      125.570000      6.890000  1.700000e+02      5.000000  
max      846.270000     46.430000  3.958244e+06      9.000000

In [4]:
# Use this space to do any additional research
# and familiarize yourself with the data.
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54639 entries, 0 to 54638
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   first        54639 non-null  object 
 1   last         54639 non-null  object 
 2   job          54639 non-null  object 
 3   phone        54639 non-null  object 
 4   email        54639 non-null  object 
 5   client_id    54639 non-null  int64  
 6   order_id     54639 non-null  int64  
 7   order_date   54639 non-null  object 
 8   order_week   54639 non-null  int64  
 9   order_year   54639 non-null  int64  
 10  item_id      54639 non-null  object 
 11  category     54639 non-null  object 
 12  subcategory  54639 non-null  object 
 13  unit_price   54639 non-null  float64
 14  unit_cost    54639 non-null  float64
 15  unit_weight  54639 non-null  float64
 16  qty          54639 non-null  int64  
 17  line_number  54639 non-null  int64  
dtypes: float64(3), int64(6), object(9)
memory usag

In [5]:
df.phone.value_counts()

phone
5016915923              220
729-620-6192x875        211
444.233.9146x233        209
265-666-3155x118        207
808.257.2029            207
                       ... 
(499)730-2617             4
+1-696-924-5100           4
+1-425-229-5573x3592      3
+1-348-322-2935           3
(506)894-3056             1
Name: count, Length: 999, dtype: int64

In [24]:
df.loc[df['first']=='Donald']

 #client_df = df.loc[df['client_id']==client_id,column]


first      last                  job               phone  \
0      Donald   Harding         Immunologist  793-904-7725x39308   
440    Donald   Harding         Immunologist  793-904-7725x39308   
1139   Donald   Simpson  Information officer        712-426-0919   
1705   Donald  Ferguson      Quality manager     +1-425-878-7810   
2008   Donald  Ferguson      Quality manager     +1-425-878-7810   
...       ...       ...                  ...                 ...   
53295  Donald   Simpson  Information officer        712-426-0919   
53385  Donald   Simpson  Information officer        712-426-0919   
53828  Donald   Harding         Immunologist  793-904-7725x39308   
53924  Donald   Harding         Immunologist  793-904-7725x39308   
54228  Donald   Simpson  Information officer        712-426-0919   

                                  email  client_id  order_id  order_date  \
0      harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
440    harding.donald.7185@sullivan.com      58515   6175632  2023-04-27   
1139   simpsonhouse@sanders-bullock.com      20875   6957099  2023-05-02   
1705       fergusond750@duarte-ryan.com      66243   7522436  2023-05-17   
2008       fergusond750@duarte-ryan.com      66243   7522436  2023-05-17   
...                                 ...        ...       ...         ...   
53295  simpsonhouse@sanders-bullock.com      20875   6510802  2023-01-08   
53385  simpsonhouse@sanders-bullock.com      20875   3265323  2023-04-15   
53828  harding.donald.7185@sullivan.com      58515   6175632  2023-04-27   
53924  harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
54228  simpsonhouse@sanders-bullock.com      20875   9965475  2023-04-02   

       order_week  order_year  ... unit_weight  qty line_number  \
0              17        2023  ...        7.50  105           1   
440            17        2023  ...        1.29   69           7   
1139           18        2023  ...        0.54   30           1   
1705           20        2023  ...       29.12   22           7   
2008           20        2023  ...        6.64  145           5   
...           ...         ...  ...         ...  ...         ...   
53295           1        2023  ...        1.86   31           1   
53385          15        2023  ...       10.46   32           5   
53828          17        2023  ...        3.28   11           6   
53924          17        2023  ...       23.82   28           0   
54228          13        2023  ...        1.89   42           3   

       line_subtotal  total_weight  shipping_price     sales_tax  \
0          115164.00        787.50         5512.50  11162.576250   
440          4180.71         89.01          623.07    444.349650   
1139         4170.90         16.20          162.00    400.793250   
1705          735.02        640.64         4484.48    482.803750   
2008        11778.35        962.80         6739.60   1712.910375   
...              ...           ...             ...           ...   
53295       14300.30         57.66          403.62   1360.112600   
53385         178.88        334.72         2343.04    233.277600   
53828        1405.03         36.08          360.80    163.339275   
53924         320.60        666.96         4668.72    461.512100   
54228        5938.80         79.38          555.66    600.737550   

          line_price  line_cost   line_profit  
0      131839.076250   85597.05  46242.026250  
440      5248.129650    4026.84   1221.289650  
1139     4733.693250    3642.30   1091.393250  
1705     5702.303750    5074.52    627.783750  
2008    20230.860375   16776.50   3454.360375  
...              ...        ...           ...  
53295   16064.032600    8089.76   7974.272600  
53385    2755.197600    2440.64    314.557600  
53828    1929.169275    1370.27    558.899275  
53924    5450.832100    4866.40    584.432100  
54228    7095.197550    6097.98    997.217550  

[145 rows x 25 columns]

In [7]:
# What three item categories had the most entries?
category_counts = df['category'].value_counts()
top_three_categories = category_counts.head(3)
top_three_categories

category
consumables    23538
furniture      11915
software        8400
Name: count, dtype: int64

In [8]:
# For the category with the most entries, which subcategory had the most entries?
most_common_category = category_counts.idxmax()
filtered_data = df[df['category'] == most_common_category]
subcategory_counts = filtered_data['subcategory'].value_counts()
top_subcategory=subcategory_counts.head(1)
top_subcategory

subcategory
bathroom supplies    6424
Name: count, dtype: int64

In [9]:
# Which five clients had the most entries in the data?
client_counts = df['client_id'].value_counts()
top_five_clients = client_counts.head(5)
top_five_clients

client_id
33615    220
66037    211
46820    209
38378    207
24741    207
Name: count, dtype: int64

In [10]:
# Store the client ids of those top 5 clients in a list.
top_five_clients.index.tolist()

top_five_client_ids = client_counts.head(5).index.tolist()
top_five_client_ids

[33615, 66037, 46820, 38378, 24741]

In [11]:
top_client_id = client_counts.idxmax()

# Filter the dataset for entries by this client
top_client_data = df[df['client_id'] == top_client_id]
top_client_data 
# Sum the total units (qty) ordered by this client
##total_units_ordered = top_client_data['qty'].sum()
##total_units_ordered

first   last                                           job  \
5      Jessica  Reyes  Production designer, theatre/television/film   
12     Jessica  Reyes  Production designer, theatre/television/film   
724    Jessica  Reyes  Production designer, theatre/television/film   
840    Jessica  Reyes  Production designer, theatre/television/film   
1405   Jessica  Reyes  Production designer, theatre/television/film   
...        ...    ...                                           ...   
52678  Jessica  Reyes  Production designer, theatre/television/film   
53450  Jessica  Reyes  Production designer, theatre/television/film   
53678  Jessica  Reyes  Production designer, theatre/television/film   
53995  Jessica  Reyes  Production designer, theatre/television/film   
54301  Jessica  Reyes  Production designer, theatre/television/film   

            phone                     email  client_id  order_id  order_date  \
5      5016915923  jessica.reyes@grimes.net      33615   1327730  2023-05-02   
12     5016915923  jessica.reyes@grimes.net      33615   6059361  2023-05-16   
724    5016915923  jessica.reyes@grimes.net      33615   7417275  2023-05-24   
840    5016915923  jessica.reyes@grimes.net      33615   1951089  2023-02-15   
1405   5016915923  jessica.reyes@grimes.net      33615   5615970  2023-04-19   
...           ...                       ...        ...       ...         ...   
52678  5016915923  jessica.reyes@grimes.net      33615   3699596  2023-01-10   
53450  5016915923  jessica.reyes@grimes.net      33615   9985826  2023-02-15   
53678  5016915923  jessica.reyes@grimes.net      33615   1630456  2023-02-24   
53995  5016915923  jessica.reyes@grimes.net      33615   2196755  2023-03-08   
54301  5016915923  jessica.reyes@grimes.net      33615   7417275  2023-05-24   

       order_week  order_year         item_id     category        subcategory  \
5              18        2023  BMB23158-88-9A        decor             plants   
12             20        2023  BDC65078-60-7M  consumables     paper products   
724            21        2023  ZBZ76765-83-2X  electronics            telecom   
840             7        2023  XBU40041-09-5M  consumables     paper products   
1405           16        2023  UCZ01380-66-1U  consumables     paper products   
...           ...         ...             ...          ...                ...   
52678           2        2023  AZB92583-19-5D  electronics            telecom   
53450           7        2023  BAA80466-20-9X  consumables  bathroom supplies   
53678           8        2023  ADC24437-65-2E  consumables               pens   
53995          10        2023  DMA18077-14-5B  consumables     paper products   
54301          21        2023  CCB33677-08-9C  consumables               misc   

       unit_price  unit_cost  unit_weight  qty  line_number  
5          172.84     154.44         2.79   78            0  
12         229.74     198.38        27.51  568            2  
724        222.03     155.57         1.33   26            2  
840          9.09       7.18         3.11   36            3  
1405        34.82      25.80         2.26   17            4  
...           ...        ...          ...  ...          ...  
52678       41.42      30.02         1.10   83            5  
53450       44.52      29.36         1.30  122            3  
53678      180.39     166.37         1.45   31            0  
53995       16.53      12.82        11.84  663            1  
54301       15.56      13.96         3.32   18            0  

[220 rows x 18 columns]

In [12]:
# How many total units (the qty column) did the client with the most entries order order?
top_client_id = client_counts.idxmax()
# Filter the dataset for entries by this client
top_client_data = df[df['client_id'] == top_client_id]
# Sum the total units (qty) ordered by this client
total_units_ordered = top_client_data['qty'].sum()
total_units_ordered

64313

## Part 2: Transform the Data
Do we know that this client spent the more money than client 66037? If not, how would we find out? Transform the data using the steps below to prepare it for analysis.

In [13]:
# Create a column that calculates the subtotal for each line using the unit_price and the qty
df['line_subtotal'] = df['unit_price'] * df['qty']

# Display the first few rows to confirm the new column
df[['unit_price', 'qty', 'line_subtotal']].head()




unit_price  qty  line_subtotal
0     1096.80  105      115164.00
1       24.95   21         523.95
2       13.52   39         527.28
3       36.42   29        1056.18
4      195.10   20        3902.00

In [14]:
# Create a column for shipping price.
# Assume a shipping price of $7 per pound for orders over 50 pounds and $10 per pound for items 50 pounds or under.

df['total_weight'] = df['unit_weight']*df['qty']
#df['total_weight']
df['shipping_price'] = df['total_weight'].apply(
    lambda weight: weight  * 7 if weight > 50 else weight * 10
)

# Display the first few rows to confirm the new column
df[['unit_price', 'unit_weight','qty','total_weight','shipping_price']].head()





unit_price  unit_weight  qty  total_weight  shipping_price
0     1096.80         7.50  105        787.50         5512.50
1       24.95         1.49   21         31.29          312.90
2       13.52         1.68   39         65.52          458.64
3       36.42         1.23   29         35.67          356.70
4      195.10        46.43   20        928.60         6500.20

In [15]:
# Create a column for the total price using the subtotal and the shipping price along with a sales tax of 9.25%

df['sales_tax'] = ((df['line_subtotal'] + df['shipping_price'])*.0925)
df['line_price'] = df['line_subtotal'] + df['shipping_price'] + df['sales_tax']
df['line_price']

df[['line_subtotal','shipping_price','line_price']].head()



line_subtotal  shipping_price     line_price
0      115164.00         5512.50  131839.076250
1         523.95          312.90     914.258625
2         527.28          458.64    1077.117600
3        1056.18          356.70    1543.571400
4        3902.00         6500.20   11364.403500

In [16]:
# Create a column for the cost of each line using unit cost, qty, and
# shipping price (assume the shipping cost is exactly what is charged to the client).

df['line_cost'] = (df['unit_cost'] * df['qty']) + df['shipping_price']
df['line_cost'] 
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ... unit_cost unit_weight  qty  line_number  \
0          17        2023  ...    762.71        7.50  105            1   
1          20        2023  ...     15.09        1.49   21            0   
2           4        2023  ...      7.86        1.68   39            6   
3          17        2023  ...     24.85        1.23   29            3   
4          21        2023  ...    108.17       46.43   20            1   

   line_subtotal  total_weight  shipping_price     sales_tax     line_price  \
0      115164.00        787.50         5512.50  11162.576250  131839.076250   
1         523.95         31.29          312.90     77.408625     914.258625   
2         527.28         65.52          458.64     91.197600    1077.117600   
3        1056.18         35.67          356.70    130.691400    1543.571400   
4        3902.00        928.60         6500.20    962.203500   11364.403500   

   line_cost  
0   85597.05  
1     629.79  
2     765.18  
3    1077.35  
4    8663.60  

[5 rows x 24 columns]

In [17]:
# Create a column for the profit of each line using line cost and line price

df['line_profit'] = df['line_price'] - df['line_cost']
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ... unit_weight  qty line_number  line_subtotal  \
0          17        2023  ...        7.50  105           1      115164.00   
1          20        2023  ...        1.49   21           0         523.95   
2           4        2023  ...        1.68   39           6         527.28   
3          17        2023  ...        1.23   29           3        1056.18   
4          21        2023  ...       46.43   20           1        3902.00   

   total_weight  shipping_price     sales_tax     line_price  line_cost  \
0        787.50         5512.50  11162.576250  131839.076250   85597.05   
1         31.29          312.90     77.408625     914.258625     629.79   
2         65.52          458.64     91.197600    1077.117600     765.18   
3         35.67          356.70    130.691400    1543.571400    1077.35   
4        928.60         6500.20    962.203500   11364.403500    8663.60   

    line_profit  
0  46242.026250  
1    284.468625  
2    311.937600  
3    466.221400  
4   2700.803500  

[5 rows x 25 columns]

## Part 3: Confirm your work
You have email receipts showing that the total prices for 3 orders. Confirm that your calculations match the receipts. Remember, each order has multiple lines.

Order ID 2742071 had a total price of \$152,811.89

Order ID 2173913 had a total price of \$162,388.71

Order ID 6128929 had a total price of \$923,441.25


In [18]:
# Check your work using the totals above
df['line_price'] = df['line_subtotal'] + df['shipping_price'] + df['sales_tax']
order_totals = df.groupby('order_id')['line_price'].sum()
order_id = (2742071,2173913,6128929)
#order_total = df[df['order_id'] == order_id]['total_price'].sum()
#order_total = df[df['order_id'].isin(order_id)]['total_price'].sum()
order_totals = df[df['order_id'].isin(order_id)].groupby('order_id')['line_price'].sum()
order_totals
for order_id, total in order_totals.items():
    print(f"Order  {order_id}: Total ${total:.2f}")

Order  2173913: Total $162388.72
Order  2742071: Total $152811.90
Order  6128929: Total $923441.24


## Part 4: Summarize and Analyze
Use the new columns with confirmed values to find the following information.

In [19]:
# How much did each of the top 5 clients by quantity spend? Check your work from Part 1 for client ids.
top_five_client_ids
client_qty_spend = df.groupby('client_id')['line_price'].sum()
client_qty_spend 
client_total = df[df['client_id'].isin(top_five_client_ids)].groupby('client_id')['line_price'].sum()
client_total
for client_id, total in client_qty_spend.items():
    if client_id in top_five_clients.index:
        print(f"Client ID {client_id}: ${total:.2f}")



#order_id = (2742071,2173913,6128929)
#order_total = df[df['order_id'] == order_id]['total_price'].sum()
#order_total = df[df['order_id'].isin(order_id)]['total_price'].sum()
#order_totals = df[df['order_id'].isin(order_id)].groupby('order_id')['total_price'].sum()
#order_totals
#for order_id, total in order_totals.items():
 #   print(f"Order  {order_id}: Total ${total:.2f}")"""





Client ID 24741: $82268891.98
Client ID 33615: $8377308.57
Client ID 38378: $12906550.88
Client ID 46820: $9743794.32
Client ID 66037: $10259514.80


In [20]:
# Create a summary DataFrame showing the totals for the for the top 5 clients with the following information:
# total units purchased, total shipping price, total revenue, and total profit. 
# top_five_client_ids
#df
#df[['client_id','qty','shipping_price','line_price','line_cost','line_profit']].head()
#summary_df=df[df['client_id'].isin(top_five_client_ids)]
#result_max=summary_df[['client_id','qty','shipping_price','line_price','line_cost','line_profit']].groupby('client_id')

def client_totals(client_id,column):
    client_df = df.loc[df['client_id']==client_id,column]
    return round(client_df.sum(),2)

data = {'client_id':top_five_client_ids}
columns = ['qty','shipping_price','line_price','line_cost','line_profit']

for col in columns:
    data[col]=[client_totals(client_id,col) for client_id in top_five_client_ids]

summary_df = pd.DataFrame(data)
summary_df

# summary_df = df.groupby('client_id').agg(
#     total_units_purchased=('qty', 'sum'),
#     total_shipping_price=('shipping_price', 'sum'),
#     total_revenue=('line_price', 'sum'),
#     total_profit=('line_profit', 'sum')
# ).reset_index()

# top_5_clients = summary_df.nlargest(5, 'total_units_purchased')

# top_5_clients
#max_df=df[df['client_id'].isin(top_five_client_ids)].groupby('client_id')



#summary = df.groupby('top_five_client_ids').agg(
 #   total_units=('units_purchased', 'sum'),
  #  total_shipping_price=('shipping_price', 'sum'),
   # total_revenue=('line_price', 'sum'),
    #total_profit=('line_profit', 'sum')
#).reset_index()

#summary_df

#df.loc[df['client_id'] == 33615].groupby



#merged_df = pd.merge(df1, df2, on='id', how='inner')

#summary_df = pd.merge(df,top_five_client_ids,on='client_id', how='inner' )
#df_summary = df[['client_id','qty','shipping_price','line_price','line_cost','line_profit']].head()
#df_summary = df.join(top_five_client_ids('client_id'),on='client_id')

#df_joined = df1.join(df2.set_index('key'), on='key')








client_id     qty  shipping_price   line_price    line_cost  line_profit
0      33615   64313      1828984.89   8377308.57   6175313.91   2201994.66
1      66037   43018      1395151.85  10259514.80   7004482.98   3255031.82
2      46820   75768      1601448.84   9743794.32   7007191.64   2736602.68
3      38378   73667      3429455.40  12906550.88   9634720.98   3271829.90
4      24741  239862      5126448.37  82268891.98  45688899.71  36579992.27

In [21]:
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number', 'line_subtotal', 'total_weight', 'shipping_price',
       'sales_tax', 'line_price', 'line_cost', 'line_profit'],
      dtype='object')

In [22]:
# Format the data and rename the columns to names suitable for presentation.

# Define the money columns. 
money_columns = ['line_price','shipping_price', 'line_cost','line_profit']
# Define a function that converts a dollar amount to millions.
def currency_format_millions(value):
    return value / 1_000_000
# Apply the currency_format_millions function to only the money columns. 
for col in money_columns:
    summary_df[col] = summary_df[col].apply(currency_format_millions)
# Rename the columns to reflect the change in the money format. 
formatted_column_names = {
  'client_id': 'Client ID',
  'line_price': 'Total Revenue (Millions)',
  'shipping_price': 'Shipping(Millions)',
  'qty': 'Units',
  'line_cost': 'Total Cost (Millions)',
  'line_profit' :'Total Profit(Millions)'
}

summary_df = summary_df.rename(columns=formatted_column_names)

summary_df




Client ID   Units  Shipping(Millions)  Total Revenue (Millions)  \
0      33615   64313            1.828985                  8.377309   
1      66037   43018            1.395152                 10.259515   
2      46820   75768            1.601449                  9.743794   
3      38378   73667            3.429455                 12.906551   
4      24741  239862            5.126448                 82.268892   

   Total Cost (Millions)  Total Profit(Millions)  
0               6.175314                2.201995  
1               7.004483                3.255032  
2               7.007192                2.736603  
3               9.634721                3.271830  
4              45.688900               36.579992

In [23]:
# Sort the updated data by "Total Profit (millions)" form highest to lowest and assign the sort to a new DatFrame.
summary_df = summary_df.sort_values('Total Profit(Millions)',ascending = False)

summary_df

Client ID   Units  Shipping(Millions)  Total Revenue (Millions)  \
4      24741  239862            5.126448                 82.268892   
3      38378   73667            3.429455                 12.906551   
1      66037   43018            1.395152                 10.259515   
2      46820   75768            1.601449                  9.743794   
0      33615   64313            1.828985                  8.377309   

   Total Cost (Millions)  Total Profit(Millions)  
4              45.688900               36.579992  
3               9.634721                3.271830  
1               7.004483                3.255032  
2               7.007192                2.736603  
0               6.175314                2.201995

Analysing the Total Profit (millions)  understood client Id #24741 earned highest profit of /$36.579992 millions,followed by  4 client Id's  38378, 66037,46820 and 33615.
In which Order ID 6128929 had a total price of /$923,441.25
Per the Report on Quantity spend by clients  Client ID 66037 spent /$10259514.80.